src/screens/SettingsScreen.js

```diff
@@ -18,11 +25,111 @@ import {
 } from '../components';
 import { SettingsListItem } from '../components/SettingsListItem';
 import { arrowLeft } from '../icons';
+import { createMatomoUserId, removeMatomoUserId, storageHelper } from '../helpers';
+import { useMatomoTrackScreenView } from '../hooks';
+
+const { MATOMO_TRACKING } = consts;
 
 export const SettingsScreen = ({ navigation }) => {
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const { orientation, dimensions } = useContext(OrientationContext);
   const [refreshing, setRefreshing] = useState(false);
+  const [sectionedData, setSectionedData] = useState([]);
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.SETTINGS);
+
+  useEffect(() => {
+    const updateSectionedData = async () => {
+      const { consent: matomoValue } = await storageHelper.matomoSettings();
+
+      setSectionedData([
+        {
+          data: [
+            {
+              title: texts.settingsTitles.pushNotifications,
+              topDivider: true,
+              type: 'toggle',
+              value: false
+            }
+          ]
+        },
+        {
+          data: [
+            {
+              title: texts.settingsTitles.analytics,
+              topDivider: true,
+              type: 'toggle',
+              value: matomoValue,
+              onActivate: (revert) =>
```

Comment:
the way onActivate and inRevert are set up, they are not async.

we do however use `await` when we call them inside of `toggleSwitch` in `SettingsListItem.js`

Reply:
you mean `onActivate` and `onDeactivate`? and your advice is to update them here?

Reply:
opening an alert does not return a promise.

it is a synchronous function with void as return value.

however, inside of SettingsListItem we call it with `await` which does imply it is async

Reply:
when i look at it now, they maybe do not need to be awaited there

Reply:
removed `await` in https://github.com/ikuseiGmbH/smart-village-app-app/pull/125/commits/caa3d385adcf707c70b62b5b1e6d15099760a06a

we need to pay attention on merging, because it needs to be done in a later case in `ToggleListItem` as well, as the file got split.